controller/apps_kubeclient.go

```diff
@@ -0,0 +1,717 @@
+	envStats, err := kc.getDeploymentEnvStats(envNS, rc)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &app.SimpleDeployment{ // TODO UUID, add version
+		Name:  &envName,
+		Stats: envStats,
+	}
+	return result, nil
+}
+
+// GetEnvironments retrieves information on all environments in the cluster
+// for the current user
+func (kc *KubeClient) GetEnvironments() ([]*app.SimpleEnvironment, error) {
+	var envs []*app.SimpleEnvironment
+	for envName := range kc.envMap {
+		env, err := kc.GetEnvironment(envName)
+		if err != nil {
+			return nil, err
+		}
+		envs = append(envs, env)
+	}
+	return envs, nil
+}
+
+// GetEnvironment returns information on an environment with the provided name
+func (kc *KubeClient) GetEnvironment(envName string) (*app.SimpleEnvironment, error) {
+	envNS, pres := kc.envMap[envName]
+	if !pres {
+		return nil, errors.New("Unknown environment: " + envName)
+	}
+
+	envStats, err := kc.getResourceQuota(envNS)
+	if err != nil {
+		return nil, err
+	}
+
+	env := &app.SimpleEnvironment{
+		Name:  &envName, // TODO UUID
+		Quota: envStats,
+	}
+	return env, nil
+}
+
+func getMetricsURLFromAPIURL(apiURLStr string) (string, error) {
+	// Parse as URL to give us easy access to the hostname
+	apiURL, err := url.Parse(apiURLStr)
+	if err != nil {
+		return "", err
+	}
+
+	// Get the hostname (without port) and replace api prefix with metrics
+	apiHostname := apiURL.Hostname()
+	if !strings.HasPrefix(apiHostname, "api") {
+		return "", errors.New("Cluster URL does not begin with \"api\": " + apiHostname)
+	}
+	metricsHostname := strings.Replace(apiHostname, "api", "metrics", 1)
+	// Construct URL using just scheme from API URL and metrics hostname
+	metricsURL := url.URL{
+		Scheme: apiURL.Scheme,
+		Host:   metricsHostname,
+	}
+	return metricsURL.String(), nil
+}
+
+func (kc *KubeClient) getDeploymentEnvStats(envNS string, rc types.UID) (*app.EnvStats, error) {
+	// Get all pods created by this deployment
+	pods, err := kc.getPods(envNS, rc)
+	if err != nil {
+		return nil, err
+	}
+	// Get the status of each pod in the deployment
+	podStats, err := kc.getPodStatus(pods)
+	if err != nil {
+		return nil, err
+	}
+
+	cpuUsage, _, err := kc.metrics.getCPUMetrics(pods, envNS) // TODO use timestamp
+	if err != nil {
+		return nil, err
+	}
+	cpuUsageInt32 := int(cpuUsage)
+	memoryUsage, _, err := kc.metrics.getMemoryMetrics(pods, envNS) // TODO use timestamp
+	if err != nil {
+		return nil, err
+	}
+	memoryUsageInt32 := int(memoryUsage)
+
+	unitsBytes := "bytes"
+	result := &app.EnvStats{
+		Cpucores: &app.EnvStatCores{
+			Used: &cpuUsageInt32,
+		},
+		Memory: &app.EnvStatMemory{
+			Used:  &memoryUsageInt32,
+			Units: &unitsBytes,
+		},
+		Pods: podStats,
+	}
+	return result, nil
+}
+
+func (kc *KubeClient) getBuildConfigs(space string) ([]string, error) {
+	// BuildConfigs are OpenShift objects, so access REST API using HTTP directly until
+	// there is a Go client for OpenShift
+	const spaceLabel string = "space"
```

Comment:
the constant should be outside of the method

Reply:
the constant has been removed.